### PR TITLE
API geolocation

### PR DESCRIPTION
**Pull request by arunbhalli**
Pt : https://www.pivotaltracker.com/story/show/178007118

### User story
```
As an API
To be able to show movies not from the visitors current location
We need to be able to locate user geolocation
```
### Changes proposed in this pull request: 
* Get visitor's latitude and longitude from client
*  Give list of 10 movies that are not available in visitor's country 

###  What I have learned working on this feature: 
* How to use Geocoder gem
*  include method 
*  how to use has_key?
*  how to use begin and rescue

### CODE REVIEW

File: .vscode/extensions.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "recommendations": []
```
Comment (offcast-dev) [inline .vscode/extensions.json:2]: It is some vscode config, but what does it do? Is it necessary?

File: app/controllers/api/movies_controller.rb [hunk 2]
```diff
@@ -3,11 +3,24 @@ def index
...
+        country = country_results.first.country
+
+        response = RestClient.get('https://unogsng.p.rapidapi.com/search?type=movie&orderby=rating',
+                                  headers = { 'x-rapidapi-key': netflix })
+        movie_array = JSON.parse(response.body)['results']
+
+        movie_results = movie_array.select { |movie| !movie['clist'].include?(country) }
+        render json: { body: movie_results[0..9]}, status: 200
+
+      elsif response = RestClient.get('https://unogsng.p.rapidapi.com/search?type=movie&orderby=rating',
```
Comment (offcast-dev) [inline app/controllers/api/movies_controller.rb:17]: I think you can take RestClient call out of if .. else statement

File: app/controllers/api/movies_controller.rb [hunk 3]
```diff
@@ -3,11 +3,24 @@ def index
...
+                                      headers = { 'x-rapidapi-key': netflix })
+        results = JSON.parse(response)['results'].select { |film| film['avgrating'] > 4 }
```
Comment (offcast-dev) [inline app/controllers/api/movies_controller.rb:19]: You don't filter for rating if the condition is true, could be a poblem

**Review by offcast-dev — CHANGES_REQUESTED** [review]
If statement should be fixed

File: .vscode/extensions.json (hunk 1 above)
Comment (lukeperrin10) [reply .vscode/extensions.json:2]: created automatically, if you want to add certain formatters or debuggers to the application

File: app/controllers/api/movies_controller.rb (hunk 2 above)
Comment (lukeperrin10) [reply app/controllers/api/movies_controller.rb:17]: done!

File: app/controllers/api/movies_controller.rb (hunk 3 above)
Comment (lukeperrin10) [reply app/controllers/api/movies_controller.rb:19]: Done, added now to the code.

**Review by offcast-dev — APPROVED** [review]

File: spec/rails_helper.rb [hunk 4]
```diff
@@ -7,6 +7,7 @@
 abort('The Rails environment is running in production mode!') if Rails.env.production?
 require 'rspec/rails'
 require 'webmock/rspec'
+WebMock.disable_net_connect!
```
Comment (1CIM) [inline spec/rails_helper.rb:10]: Can you explain this as well?

File: app/controllers/api/movies_controller.rb [hunk 5]
```diff
@@ -3,11 +3,24 @@ def index
     netflix = Rails.application.credentials.unogsng_key
 
     begin
-      response = RestClient.get('https://unogsng.p.rapidapi.com/search?type=movie&orderby=rating',
-                                headers = { 'x-rapidapi-key': netflix })
-      results = JSON.parse(response)['results'].select { |film| film['avgrating'] > 4 }
-      netflix_sorted = results.sort_by { |film| film['avgrating'] }.reverse
-      render json: { body: netflix_sorted[0..9] }, status: 200
+      if params.has_key?(:lat) && params.has_key?(:lon)
```
Comment (1CIM) [inline app/controllers/api/movies_controller.rb:6]: This has_key thing. would you like to explain to me how it works/dose?

File: spec/rails_helper.rb (hunk 4 above)
Comment (lukeperrin10) [reply spec/rails_helper.rb:10]: Yeah, during the testing we do not want to contact the actual internet, so we want to disable the connection. However we can during manual testing enable this to be able to see our requests are working correctly.

Comment (1CIM) [reply spec/rails_helper.rb:10]: i see 👍

File: app/controllers/api/movies_controller.rb (hunk 5 above)
Comment (lukeperrin10) [reply app/controllers/api/movies_controller.rb:6]: it bascially acts as a boolean, it checks the params sent with the request. If Lat and Lon are present then returns true and continues in that if statement. If false continues to the elsif statement.

Comment (1CIM) [reply app/controllers/api/movies_controller.rb:6]: 👍👍

**Pull request merged**